src/main/java/bio/terra/service/filedata/azure/util/BlobContainerClientFactory.java

```diff
@@ -1,90 +1,71 @@
 package bio.terra.service.filedata.azure.util;
 
+import static bio.terra.common.ValidationUtils.requiresNotBlank;
+
 import com.azure.core.credential.AzureSasCredential;
 import com.azure.core.credential.TokenCredential;
 import com.azure.core.http.HttpClient;
-import com.azure.storage.blob.BlobClient;
 import com.azure.storage.blob.BlobContainerClient;
 import com.azure.storage.blob.BlobContainerClientBuilder;
 import com.azure.storage.blob.BlobServiceClient;
 import com.azure.storage.blob.BlobServiceClientBuilder;
 import com.azure.storage.blob.BlobUrlParts;
-import com.azure.storage.blob.models.UserDelegationKey;
-import com.azure.storage.blob.sas.BlobSasPermission;
-import com.azure.storage.blob.sas.BlobServiceSasSignatureValues;
 import com.azure.storage.common.StorageSharedKeyCredential;
-import com.azure.storage.common.sas.SasProtocol;
-import java.time.OffsetDateTime;
-import java.time.ZoneOffset;
+import java.time.Duration;
 import java.util.Locale;
 import java.util.Objects;
-import org.apache.commons.lang3.StringUtils;
 
 /**
  * A class that wraps the SDK container client and facilities the creation of SAS Urls for blobs by
  * abstracting the different strategies that depend on the authentication mechanism.
  */
 public class BlobContainerClientFactory {
 
-  public static final int EXPIRATION_DAYS = 7;
+  public static final Duration DELEGATED_KEY_EXPIRATION = Duration.ofHours(24);
   private final HttpClient httpClient = HttpClient.createDefault();
 
   private final BlobContainerClient blobContainerClient;
 
-  private final SasTokenGeneratorStrategy sasTokenGeneratorStrategy;
-  private final AzureSasCredential blobContainerSasTokenCreds;
-  private final BlobServiceClient blobServiceClient;
-  private UserDelegationKey delegationKey;
-
-  public BlobContainerClientFactory(String accountName, String accountKey, String containerName) {
-
-    if (StringUtils.isBlank(accountName)) {
-      throw new IllegalArgumentException("The account name is either null or empty.");
-    }
+  public BlobSasUrlFactory getBlobSasUrlFactory() {
```

Comment:
It looks like the way this class is used is `containerClientFactory.getBlobSasUrlFactory().something()`. Is there a need for any caller to use `getBlobSasUrlFactory()` except to call another method? If not, a nicer API would be to put `something()` directly on this class, so callers can omit the `getBlobSasUrlFactory()` method call.

Reply:
The high level API is the `BlobCrl` there you will find the method for `something()` directly.

Reply:
In case this wasn't clear, what I'm saying is add methods to this class so the call site:
```
    return this.sourceClientFactory
        .getBlobSasUrlFactory()
        .createSasUrlForBlob(blobName, blobSasTokenOptions);
```
can be written as
```
    return this.sourceClientFactory
        .createSasUrlForBlob(blobName, blobSasTokenOptions);
```
Since a caller shouldn't need to know that a `BlobSasUrlFactory` is used to generate the response to `createSasUrlForBlob()`.

Reply:
Yep, I understood that. The point I tried to make is that callers should use the BlobCrl API to create URLs with SAS tokens.  The BlobCrl class has the method`createSasTokenUrlForBlob`. Before this PR the `BlobClientContainerFactory` had this method, and it seems unnecessary to duplicate it there and further complicate the role/purpose of the `BlobClientContainerFactory`.

Reply:
I don't think this change complicates the purpose of the class; to me hiding this detail makes its role clearer. If you have a getter, and its only use is to call a method on the object returned, this is effectively the same as making the enclosed object's API part of the owning object's API. Getting rid of the getter makes this relationship clearer. It's true that since this is only used by `BlobCrl` it doesn't affect general user code.

I don't think there's duplication here. I'm suggesting you replace:

```
  public BlobSasUrlFactory getBlobSasUrlFactory() {
    return blobSasUrlFactory;
  }
```

with 

```
  public String createSasUrlForBlob(String blobName, BlobSasTokenOptions options) {
    return blobSasurlFactory.createSasurlForBlob(blobName, options);
  }
```
You could make `BlobContainerClientFactory implement BlobSasUrlFactory`  too. To me this kind of refactoring can help make the object relationships clearer. If you disagree, I'm sure there will be a chance to revisit it in the future.

Reply:
Adding the method seems to complicate further defining a name that describes the functionality.  Another point that may not be obvious is that the `BlobContainerClientFactory` exposes the blob container client via the getter. So the caller's expected interaction with the properties directly is consistent.  As it stands right now, the BlobConainerClientFactory is responsible for wiring up two properties (The blob container client and the SAS URL factory). Maybe a better name for the class could be `BlobContaienrClientDao` -  which hopefully better represents its function/role.